Update to AI-based tooling policy (#6477)
carbon-language/carbon-lang: CONTRIBUTING.md

#### Using AI-based contribution tools

All submissions to Carbon need to follow our
[Contributor License Agreement (CLA)](#contributor-license-agreements-clas),
which covers any original work of authorship included in the submission. This

> doesn’t prohibit the use of coding assistance tools, including tool-, AI-, or

Reviewer: ```suggestion
doesn't prohibit the use of coding assistance tools, including tool-, AI-, or
```
May as well use an ASCII apostrophe here for consistency.

requirements.

All contributions, regardless of what tools are used, are also still the
responsibility of the operator of these tools and subject to normal code review
and our [guidelines and standards](#contribution-guidelines-and-standards)
below. We also emphasize two additional requirements for contributors operating
or using AI-based tools:

1. **Contributions should not become extractive of the project and community**:
   the value added should outweigh the overhead of landing the contribution. The
   overhead of landing contributions ranges from code review, to discussions,
   distractions from the current project priorities, or growing maintenance
   burden without growing maintainers.

2. **Each PR should be transparent about the tooling used** in proportion to how
   much of the PR was produced by the tool and whether the tool is a standard
   one for the project. For example, formatting with the standard tools is
   reasonable to assume without further comment. But a PR that is largely
   derived from running a specific Python script, regular expression, or
   AI-based tool over the codebase, we ask the PR to be transparent about this.

>    Ideally, this include a prose description of how the tool was used to

Reviewer: ```suggestion
   reasonable to assume without further comment. But if a PR is largely
   derived from running a specific Python script, regular expression, or
   AI-based tool over the codebase, we ask that its description is transparent about this.
   Ideally, this should include a prose description of how the tool was used to
```
(Just massaging the grammar a bit.)

Incidentally: I think I'd prefer a "here is the exact `sed` command I ran" description over a prose description. We could maybe drop the word "prose" here?

   formulate the change. Where relevant, we also encourage following the pattern
   established by the Fedora Project to mark commits with `Assisted-by: ...`.

Our policies and practices here are inspired by and aim to be roughly compatible
with several other open source projects:

-   [Fedora Project's policy](https://docs.fedoraproject.org/en-US/council/policy/ai-contribution-policy/)
-   [LLVM Developer Policy around AI generated code](https://llvm.org/docs/DeveloperPolicy.html#ai-generated-contributions)

As the open source community evolves and learns how best to integrate these

> tools into the project and development workflows, we expect to reflect that with

Reviewer 2: ```suggestion
tools into project and development workflows, we expect to reflect that with
```

   one for the project. For example, formatting with the standard tools is
   reasonable to assume without further comment. But if a PR is largely derived
   from running a specific Python script, regular expression, or AI-based tool

>    over the codebase, we ask that its description is transparent about this and

Reviewer: Optional: replace the first "description" with something like "commit message" to avoid repeating "description" twice in this sentence.

Author: Done.
